contracts/its-stacks-translator/src/contract/its_hub_message_to_clarity_bytes.rs

```diff
@@ -0,0 +1,342 @@
+use axelar_wasm_std::nonempty;
+use axelar_wasm_std::nonempty::Uint256;
+use clarity::vm::representations::ClarityName;
+use clarity::vm::types::{TupleData, Value};
+use cosmwasm_std::{HexBinary, Uint128};
+use interchain_token_service_std::{
+    DeployInterchainToken, HubMessage, InterchainTransfer, Message, TokenId,
+};
+use router_api::ChainNameRaw;
+
+use crate::error::ContractError;
+
+pub const MESSAGE_TYPE_INTERCHAIN_TRANSFER: u128 = 0;
+pub const MESSAGE_TYPE_DEPLOY_INTERCHAIN_TOKEN: u128 = 1;
+
+pub const CLARITY_NAME_TYPE: &str = "type";
+const CLARITY_NAME_SOURCE_CHAIN: &str = "source-chain";
+pub const CLARITY_NAME_TOKEN_ID: &str = "token-id";
+pub const CLARITY_NAME_SOURCE_ADDRESS: &str = "source-address";
+pub const CLARITY_NAME_DESTINATION_ADDRESS: &str = "destination-address";
+pub const CLARITY_NAME_AMOUNT: &str = "amount";
+pub const CLARITY_NAME_DATA: &str = "data";
+pub const CLARITY_NAME_NAME: &str = "name";
+pub const CLARITY_NAME_SYMBOL: &str = "symbol";
+pub const CLARITY_NAME_DECIMALS: &str = "decimals";
+const CLARITY_NAME_MINTER_BYTES: &str = "minter-bytes";
+
+pub fn its_hub_message_to_clarity_bytes(
+    its_hub_message: HubMessage,
+) -> Result<Vec<u8>, ContractError> {
+    let payload = match its_hub_message {
+        HubMessage::SendToHub { .. } => Err(ContractError::InvalidPayload),
+        HubMessage::ReceiveFromHub {
+            source_chain,
+            message,
+        } => match message {
+            Message::InterchainTransfer(InterchainTransfer {
+                token_id,
+                source_address,
+                destination_address,
+                amount,
+                data,
+            }) => get_its_interchain_transfer_payload_receive_from_hub(
+                source_chain,
+                token_id,
+                source_address.into(),
+                destination_address.into(),
+                amount,
+                data,
+            ),
+            Message::DeployInterchainToken(DeployInterchainToken {
+                token_id,
+                name,
+                symbol,
+                decimals,
+                minter,
+            }) => get_its_deploy_interchain_token_payload_receive_from_hub(
+                source_chain,
+                token_id,
+                name.into(),
+                symbol.into(),
+                decimals,
+                minter,
+            ),
+            Message::LinkToken(_) => Err(ContractError::InvalidPayload),
```

Comment:
Is this meant to be temporary? Are we not supporting linking custom tokens at launch? will we ever support this?

Reply:
can we add support for these types in the translator even if the Clarity ITS contract doesn't have support for it yet?

Reply:
The Stacks Clarity contract does not yet support this, we can add this after it is supported.
We don't know how the clarity function signature/payload for this will look like yet, so even if we implement something now, most likely it will not be correct and will require changing.